manim/mobject/geometry.py

```diff
@@ -863,6 +864,225 @@ def __init__(self, n=6, **kwargs):
         Polygon.__init__(self, *vertices, **kwargs)
 
 
+class ArcPolygon(VMobject):
+    """A generalized polygon allowing for points to be connected with arcs.
+
+    This version tries to stick close to the way :class:`Polygon` is used. Points can be
+    passed to it directly which are used to generate the according arcs
+    (using :class:`ArcBetweenPoints`). An angle or radius can be passed to it to
+    use across all arcs, but to configure arcs individually an arc_config list has to be
+    passed with the syntax explained below.
+
+    ArcPolygons can be transformed properly into one another as well. Be
+    advised that any arc initialized with 'angle=0' will actually be
+    made like a line, so if a straight secion should seamlessly
+    transform into an arced section, initialize the straight section
+    with a negligible angle instead (such as 'angle=0.0001').
+
+    There is an alternative version that is instantiated with pre-defined arcs.
+
+    See Also
+    --------
+    :class:`ArcPolygonFromArcs`
+
+    Attributes
+    ----------
+    arcs : :attr:`list`
```

Comment:
```suggestion
    arcs : :class:`list`
```

Reply:
While attributes are currently glitched in rendering, I'm pretty sure your suggestion is wrong, as per the wiki: `Attribute names should be wrapped in :attr:`path_goes_here`​. See examples in the subsection above.`

Reply:
The `:attr:` role should be used when you want to reference from somewhere else to `ArcPolygon[FromArcs].arcs`. This occurrence here specifies the type of the attribute; `:class:` is correct.

Reply:
That's a bit counterintuitive but alright, taking your word for it and commiting